File: frontend/lib/features/alumni/domain/entities/alumni.dart
import 'package:equatable/equatable.dart';
import 'batch.dart';

class Alumni extends Equatable {
  final int? id;
  final String name;
  final Batch batch;
  final String email;
  final String? phone;
  final String? occupation;
  final String? company;
  final String? address;
  final String? city;
  final String? state;
  final String? country;
  final String? bio;
  final String? linkedinUrl;
  final String? profilePhotoUrl;
  final int? graduationYear;

  const Alumni({
    this.id,
    required this.name,
    required this.batch,
    required this.email,
    this.phone,
    this.occupation,
    this.company,
    this.address,
    this.city,
    this.state,
    this.country,
    this.bio,
    this.linkedinUrl,
    this.profilePhotoUrl,
    this.graduationYear,
  });

  String get initials {
    final parts = name.trim().split(RegExp(r'\s+'));
    if (parts.length >= 2 && parts[0].isNotEmpty && parts[1].isNotEmpty) {
      return (parts[0][0] + parts[1][0]).toUpperCase();
    }
    return name.isNotEmpty ? name[0].toUpperCase() : '?';
  }

  String get location {
    final parts = <String>[];
    if (city != null && city!.isNotEmpty) parts.add(city!);
    if (state != null && state!.isNotEmpty) parts.add(state!);
    return parts.join(', ');
  }

  String get headline {
    if (occupation == null || occupation!.isEmpty) return 'Alumni';
    if (company == null || company!.isEmpty) return occupation!;
    return '$occupation at $company';
  }

  @override
  List<Object?> get props => [
        id, name, batch, email, phone, occupation, company,
        address, city, state, country, bio, linkedinUrl,
        profilePhotoUrl, graduationYear,
      ];
}

File: backend/src/main/java/com/reunion/domain/model/Alumni.java
package com.reunion.domain.model;

import java.time.LocalDateTime;
import java.util.Optional;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.ToString;

/**
 * Pure domain entity. No framework, no JPA.
 * Lombok generates the builder, equals/hashCode, and toString.
 * Required fields use @NonNull so the generated constructor enforces it.
 * Optional getters are written manually so absence is explicit at the type level.
 */
@Builder(toBuilder = true)
@EqualsAndHashCode
@ToString
public class Alumni {

    private final Long id;
    @NonNull private final String name;
    @NonNull private final Batch batch;
    @NonNull private final String email;
    private final String phone;
    private final String occupation;
    private final String company;
    private final String address;
    private final String city;
    private final String state;
    private final String country;
    private final String bio;
    private final String linkedinUrl;
    private final String profilePhotoUrl;
    private final Integer graduationYear;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;

    public Optional<Long> getId() { return Optional.ofNullable(id); }
    public String getName() { return name; }
    public Batch getBatch() { return batch; }
    public String getEmail() { return email; }
    public Optional<String> getPhone() { return Optional.ofNullable(phone); }
    public Optional<String> getOccupation() { return Optional.ofNullable(occupation); }
    public Optional<String> getCompany() { return Optional.ofNullable(company); }
    public Optional<String> getAddress() { return Optional.ofNullable(address); }
    public Optional<String> getCity() { return Optional.ofNullable(city); }
    public Optional<String> getState() { return Optional.ofNullable(state); }
    public Optional<String> getCountry() { return Optional.ofNullable(country); }
    public Optional<String> getBio() { return Optional.ofNullable(bio); }
    public Optional<String> getLinkedinUrl() { return Optional.ofNullable(linkedinUrl); }
    public Optional<String> getProfilePhotoUrl() { return Optional.ofNullable(profilePhotoUrl); }
    public Optional<Integer> getGraduationYear() { return Optional.ofNullable(graduationYear); }
    public Optional<LocalDateTime> getCreatedAt() { return Optional.ofNullable(createdAt); }
    public Optional<LocalDateTime> getUpdatedAt() { return Optional.ofNullable(updatedAt); }
}
